### PR TITLE
Adds Lorenz and gini over time plots

### PR DESCRIPTION
**Pull request by vulder**
New plots to draw the Lorenz curve for a projects IN/OUT interactions.
Add, a new plot to visualize the gini coefficient for interactions and code churn over time.

### CODE REVIEW

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/utils/git_util.py  1
- varats/plots/blame_lorenz_curve.py  9
         

Clones added
============
- varats/plots/repository_churn.py  1
- varats/plots/blame_lorenz_curve.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/VaRA/VaRA-Tool-Suite/pullRequest?prid=4908778&bid=16251832)

File: varats/data/reports/blame_report.py [hunk 1]
```diff
@@ -269,3 +270,41 @@ def generate_max_time_distribution_tuples(report: BlameReport,
...
+    Args:
+        report: BlameReport to get the interactions from
+    """
+    head_interactions = []
+    for func_entry in report.function_entries:
+        for interaction in func_entry.interactions:
+            for interacting_commit in interaction.interacting_hashes:
+                if interacting_commit.startswith(report.head_commit):
+                    head_interactions.append(interaction)
+                    continue
```
Comment (boehmseb) [inline varats/data/reports/blame_report.py]:
```suggestion
                    break
```

File: varats/plots/blame_lorenz_curve.py [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+        raise NotImplementedError
+
+
+def gini(lorenz_values: pd.Series) -> pd.Series:
+    """
+    Calculates the gini coefficient:  half of the relative mean absolute
+    difference between the lorenz values
+
+    Args:
+        lorenz_values: the scaled prefix sum of a values range, in ordered
```
Comment (boehmseb) [inline varats/plots/blame_lorenz_curve.py]: This description is a bit unclear.
Do you mean `lorenz_values: the scaled prefix sum of an ordered values range`?

File: varats/plots/blame_lorenz_curve.py [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+              label=label,
+              color='orange')
+
+
+def draw_gini_blame_over_time(axis: axes.SubplotBase, blame_data: pd.DataFrame,
+                              consider_in_interactions: bool,
+                              consider_out_interactions: bool,
+                              plot_cfg: tp.Dict[str, tp.Any]) -> None:
+    """
+    Draws the gini of the blame interactions over time.
```
Comment (boehmseb) [inline varats/plots/blame_lorenz_curve.py]:
```suggestion
    Draws the gini coefficients of the blame interactions over time.
```

**Review by boehmseb — CHANGES_REQUESTED** [review]

File: varats/plots/blame_lorenz_curve.py (hunk 2 above)
Comment (vulder) [inline varats/plots/blame_lorenz_curve.py]:
```suggestion
        lorenz_values: the scaled prefix sum of an ordered values range
```

**Review by boehmseb — APPROVED** [review]
lgtm

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/plots/blame_lorenz_curve.py  9
- varats/utils/git_util.py  1
         

Clones added
============
- varats/plots/blame_lorenz_curve.py  5
- varats/plots/repository_churn.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/VaRA/VaRA-Tool-Suite/pullRequest?prid=4908778&bid=16251832)

**Pull request merged**